### DISCUSSION TITLE
Ability to use a Tone.js instance within EnvelopePlugin

### DISCUSSION THREAD

**Discussion by corysimmons** [katspaugh/wavesurfer.js#3525] [Ideas]
EnvelopePlugin only allows you to ramp in/out volume.

It would be amazing if it could also ramp in/out Tone.js effects.

**Comment by katspaugh** [comment]
The way the Envelope plugin is currently implemented, it's just a change of volume over time using the regular volume control of the HTMLMediaElement.

If you want to use it in a Web Audio context with a library like Tone.js, you could theoretically use the draggable controls and their data points to configure your envelope audio nodes (I assume they are implemented as custom audio worklet nodes in Tone.js), then connect the wavesurfer audio output (via a MediaElementSourceNode) to a gain node and parametrize its gain with the envelopes.
